src/NHibernate/Criterion/InExpression.cs

```diff
@@ -135,15 +135,15 @@ private List<TypedValue> GetParameterTypedValues(ICriteria criteria, ICriteriaQu
 						object subval = _values[j] == null
 											? null
 											: actype.GetPropertyValues(_values[j])[i];
-						list.Add(new TypedValue(types[i], subval));
+						list.Add(new TypedValue(types[i], subval, false));
 					}
 				}
 
 				return list;
 			}
 			else
 			{
-				return _values.Select(v => new TypedValue(type, v)).ToList();
+				return _values.Select(v => new TypedValue(type, v, false)).ToList();
```

Comment:
Not sure about this...

Reply:
`_values` is the collection of values to be put in the `in` clause. They must be elements, not themselves lists.